.ado/publish.yml

```diff
@@ -5,38 +5,156 @@
 variables:
   Ver.Major: 0
   Ver.Minor: 0
-  Ver.Patch: 0
+  Ver.Patch: 4
 
 name: $(BuildId)
 
 trigger: none
 pr: none
 
 jobs:
-- job: "Publish"
+- job: "Build"
+  strategy:
+    matrix:
+      #linux:                           # 2023.06.30 ESRP Publishing of Py packs is not supported for linux_x86_64.
+      #  imageName: 'ubuntu-latest'     # TODO(kuzminrobin): Manilinux.
+      mac:
+        imageName: 'macOS-latest'       # MacOS-specific Py + VSCode Ext
+      windows:
+        imageName: 'windows-latest'     # Win-specific Py + Plkatform-independent Py
   pool: 
-    vmImage: 'ubuntu-latest'
+    vmImage: $(imageName)
+
   steps:
   - task: UsePythonVersion@0
     inputs:
       versionSpec: '3.11'
 
   - script: |
       python ./prereqs.py --install
+    displayName: Install Prereqs
+
+  - script: |
       npm install -g @vscode/vsce
-    displayName: Install VSCE Pub Prereqs
-    
+    displayName: Install Prereqs for VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+   
   - script: |
       python build.py --wasm --npm --vscode
     displayName: Build VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      python build.py --pip
+    displayName: Build Platform-Dependent Py Packages
       
+  - script: |
+      python build.py --jupyterlab
+    displayName: Build Platform-Independent Py Packages
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
   - script: |
       cp ../LICENSE.txt .
       vsce package $(Ver.Major).$(Ver.Minor).$(Ver.Patch)
     displayName: Pack VSCode Extension
     workingDirectory: '$(System.DefaultWorkingDirectory)/vscode'
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      ls vscode/*
+      ls vscode/out/*
+    displayName: List VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/qsharp-vscode-$(Ver.Major).$(Ver.Minor).$(Ver.Patch).vsix
+    artifact: VSIX
+    displayName: Upload VSCode Extension Artifact
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/out/extension.js
```

Comment:
Why does this specific artifact need to be published? Don't you only need the VSIX?

Reply:
That doesn't make sense. You shouldn't need to copy it, it should be in the VSIX. it maybe be missing in the build somehow. (I never need to copy it when publishing my test package). We should get to the bottom of this, not hack around it. Did you figure out why publishing fails? is there any specific error?

Reply:
I rebased my branch on top of today's main and double-checked.
The file is really inside of the VSIX under the path "extension\out\extension.js". 
But when publishing the extension, if the file is not in the file system at "out\extension.js" then I get the following error message:
```
 ERROR  Extension entrypoint(s) missing. Make sure these files exist and aren't ignored by '.vscodeignore':
  extension/out/extension.js
```
The file inside of the VSIX is identical to the one generated in the file system.

The similar story is with the "LICENSE.txt", but results in a warning rather than an error.

I spent some time reading through the [VSCode Extension Development docs](https://code.visualstudio.com/api) but so far didn't find anything applicable. Will try to dig in a bit deeper.